Framework/PythonInterface/mantid/plots/mantidaxes.py

```diff
@@ -1132,6 +1134,33 @@ def __init__(self, *args, **kwargs):
         # it interfering with double-clicking on the axes.
         self.figure.canvas.mpl_disconnect(self._cids[1])
 
+    def set_xlim3d(self, *args):
+        min, max = super().set_xlim3d(*args)
+
+        if hasattr(self, 'original_data'):
+            x = self.original_data[0].copy()
+            x[np.less(x, min, where=~np.isnan(x))] = np.nan
+            x[np.greater(x, max, where=~np.isnan(x))] = np.nan
+            self.collections[0]._vec[0] = x
+
+    def set_ylim3d(self, *args):
+        min, max = super().set_ylim3d(*args)
+
+        if hasattr(self, 'original_data'):
+            y = self.original_data[1].copy()
+            y[np.less(y, min, where=~np.isnan(y))] = np.nan
+            y[np.greater(y, max, where=~np.isnan(y))] = np.nan
+            self.collections[0]._vec[1] = y
+
+    def set_zlim3d(self, *args):
+        min, max = super().set_zlim3d(*args)
+
+        if hasattr(self, 'original_data'):
+            z = self.original_data[2].copy()
+            z[np.less(z, min, where=~np.isnan(z))] = np.nan
+            z[np.greater(z, max, where=~np.isnan(z))] = np.nan
+            self.collections[0]._vec[2] = z
```

Comment:
The functionality seems to work well. Do you really need these essentially identical pieces of code repeated?